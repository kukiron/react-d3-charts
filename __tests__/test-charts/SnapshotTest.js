import React from "react"
import renderer from "react-test-renderer"

import BarChart from "../../src/bar-chart"
import PieChart from "../../src/pie-chart"

describe("Snapshot of the Charts", () => {
  const data = [
    { title: "Plane Ticket", value: 640 },
    { title: "Hotel", value: 135 },
    { title: "Rental Car", value: 40 },
    { title: "Dinner", value: 55 },
    { title: "Postcards", value: 4 }
  ]

  // Snapshot of BarChart
  it("bar chart renders correctly", () => {
    function createNodeMock(element) {
      if (element.type === "div") {
        return {
          getBoundingClientRect() {
            return { width: 600 }
          }
        }
      }
      return null
    }
    const options = { createNodeMock }
    const tree = renderer.create(<BarChart data={data} />, options).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // Snapshot of PieChart
  it("pie chart renders correctly", () => {
    function createNodeMock(element) {
      if (element.type === "div") {
        return {
          getBoundingClientRect() {
            return { width: 900 }
          }
        }
      }
      return null
    }
    const options = { createNodeMock }
    const tree = renderer.create(<PieChart data={data} />, options).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
