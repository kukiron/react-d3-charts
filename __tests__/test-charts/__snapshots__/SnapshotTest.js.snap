// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Charts bar chart renders correctly 1`] = `
<div>
  <svg
    className="vizzlo-bar"
    height={550}
    width={600}
  >
    <g>
      <g
        className="axis axis-Bottom"
        transform="translate(0, 450)"
      />
      <g
        className="axis axis-Left"
        transform="translate(60, 0)"
      />
    </g>
    <g>
      <rect
        className="bar-chart"
        fill="#E74C3C"
        height={286.9918699186992}
        width={24.761904761904763}
        x={84.76190476190476}
        y={163.0081300813008}
      >
        <title>
          €353
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#3498DB"
        height={309.7560975609756}
        width={24.761904761904763}
        x={134.28571428571428}
        y={140.2439024390244}
      >
        <title>
          €381
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#2ECC71"
        height={369.10569105691053}
        width={24.761904761904763}
        x={183.8095238095238}
        y={80.89430894308947}
      >
        <title>
          €454
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#F1C40F"
        height={390.2439024390244}
        width={24.761904761904763}
        x={233.33333333333334}
        y={59.756097560975604}
      >
        <title>
          €480
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#9B59B6"
        height={315.4471544715447}
        width={24.761904761904763}
        x={282.8571428571429}
        y={134.5528455284553}
      >
        <title>
          €388
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#E74C3C"
        height={400}
        width={24.761904761904763}
        x={332.3809523809524}
        y={50}
      >
        <title>
          €492
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#3498DB"
        height={164.22764227642278}
        width={24.761904761904763}
        x={381.9047619047619}
        y={285.7723577235772}
      >
        <title>
          €202
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#2ECC71"
        height={297.5609756097561}
        width={24.761904761904763}
        x={431.42857142857144}
        y={152.4390243902439}
      >
        <title>
          €366
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#F1C40F"
        height={83.739837398374}
        width={24.761904761904763}
        x={480.95238095238096}
        y={366.260162601626}
      >
        <title>
          €103
        </title>
      </rect>
      <rect
        className="bar-chart"
        fill="#9B59B6"
        height={22.76422764227641}
        width={24.761904761904763}
        x={530.4761904761905}
        y={427.2357723577236}
      >
        <title>
          €28
        </title>
      </rect>
    </g>
  </svg>
</div>
`;

exports[`Snapshot of the Charts pie chart renders correctly 1`] = `
<div>
  <svg
    height={768}
    width={1024}
  >
    <g
      className="vizzlo-pie"
      transform="translate(512, 384)"
    >
      <g>
        <g>
          <path
            className="pie-chart"
            d="M1.6459252980540425e-14,-268.79999999999995A268.79999999999995,268.79999999999995,0,0,1,169.66275229655713,-208.48978508108513L0,0Z"
            fill="#E74C3C"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(97.40028108898876,-274.00333071659173)"
          >
            Led Zeppelin
          </text>
          <title>
            €353
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M169.66275229655713,-208.48978508108513A268.79999999999995,268.79999999999995,0,0,1,265.7634820560918,-40.2891003302548L0,0Z"
            fill="#3498DB"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(252.49412419718368,-144.26141981104007)"
          >
            AC/DC
          </text>
          <title>
            €381
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M265.7634820560918,-40.2891003302548A268.79999999999995,268.79999999999995,0,0,1,200.64818192650404,178.86795993019126L0,0Z"
            fill="#2ECC71"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(278.75609292093213,82.82318914082339)"
          >
            Def Lepard
          </text>
          <title>
            €454
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M200.64818192650404,178.86795993019126A268.79999999999995,268.79999999999995,0,0,1,-23.1179862877764,267.8040304214971L0,0Z"
            fill="#F1C40F"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(107.40630543908568,270.23790546835914)"
          >
            Van Halen
          </text>
          <title>
            €480
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M-23.1179862877764,267.8040304214971A268.79999999999995,268.79999999999995,0,0,1,-199.60643856501858,180.02974666257083L0,0Z"
            fill="#9B59B6"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(-129.49476723222043,260.37616108137286)"
          >
            Eagles
          </text>
          <title>
            €388
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M-199.60643856501858,180.02974666257083A268.79999999999995,268.79999999999995,0,0,1,-262.42758488035673,-58.18249473736199L0,0Z"
            fill="#E74C3C"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(-281.1863908475838,74.15425545516449)"
          >
            Black Sabbath
          </text>
          <title>
            €492
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M-262.42758488035673,-58.18249473736199A268.79999999999995,268.79999999999995,0,0,1,-220.46533857712902,-153.7806050387104L0,0Z"
            fill="#3498DB"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(-266.2771324097929,-116.88083142935629)"
          >
            Scorpions
          </text>
          <title>
            €202
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M-220.46533857712902,-153.7806050387104A268.79999999999995,268.79999999999995,0,0,1,-67.41194750221457,-260.2096641824793L0,0Z"
            fill="#2ECC71"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(-166.0204902140728,-238.75057451046885)"
          >
            Starship
          </text>
          <title>
            €366
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M-67.41194750221457,-260.2096641824793A268.79999999999995,268.79999999999995,0,0,1,-14.557016083079468,-268.405538845153L0,0Z"
            fill="#F1C40F"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(-44.55995432546818,-287.36570858491825)"
          >
            Chuck Berry
          </text>
          <title>
            €103
          </title>
        </g>
        <g>
          <path
            className="pie-chart"
            d="M-14.557016083079468,-268.405538845153A268.79999999999995,268.79999999999995,0,0,1,-4.937775894162127e-14,-268.79999999999995L0,0Z"
            fill="#9B59B6"
            stroke="#fff"
          />
          <text
            className="pie-label"
            dy=".35em"
            transform="translate(-7.8771096892564625,-290.6932939421606)"
          >
            Nirvana
          </text>
          <title>
            €28
          </title>
        </g>
      </g>
    </g>
  </svg>
</div>
`;
